.github/workflows/CI.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  # run once a day at noon UTC
+  schedule:
+    - cron: "0 12 * * *"
+  push:
+    branches:
+      - "master"
+      # For development remove after finishing python 3 support
+      - "python_3"
+  pull_request:
+    branches:
+      - "master"
+
+concurrency:
+  group: "${{ github.ref }}-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: Test on ${{ matrix.os }}, Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [2.7, 3.7, 3.8, 3.9]
+    env:
+      MPLBACKEND: agg
+
+    steps:
+    - uses: actions/checkout@v2
+
+    # More info on options: https://github.com/conda-incubator/setup-miniconda
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        auto-update-conda: true
+        mamba-version: "*"
+        channels: conda-forge,defaults,bioconda
+        channel-priority: true
+        show-channel-urls: true
+
+    - name: Conda info
+      shell: bash -l {0}
+      run: |
+        conda info
+        conda list
+        conda config --show-sources
+        conda config --show
+
+
+    - name: Python version
+      shell: bash -l {0}
+      run: |
+        - python -c "import sys; print(sys.version)"
+
+    - name: Install package dependencites
+      shell: bash -l {0}
+      run: |
+        python -m pip install . --no-deps
+        conda list
+
+    - name: Install pytest in and plugins
+      shell: bash -l {0}
+      run: |
+        mamba install pytest pytest-pep8 pytest-cov codecov
+
+    - name: Install GROMACS (latest)
+      shell: bash -l {0}
+      run: |
+         mamba install gromacs pocl
+
+    - name: Install package
+      shell: bash -l {0}
+      run: |
+        python -m pip install --no-deps .
+    - name: Run tests
+      shell: bash -l {0}
+      run: |
+        pytest -v --disable-pytest-warnings --durations=20 --low-performance --cov=gromacs --cov-report=xml ./tests
+    - name: Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        file: ./coverage.xml
+        fail_ci_if_error: true
```

Comment:
Started CI file similar to one by @orbeckst for gromacswrapper, work in progress.